docs/foam-local-plugins.md

```diff
@@ -0,0 +1,25 @@
+# Foam Local Plugins
+
+Foam can use workspace plugins to provide customization for users.
+
+## Goal
+
+Here are some of the things that we could enable with local plugins in Foam:
+- extend the document syntax to support roam style attributes (e.g. `stage:: seedling`)
+- automatically add tags to my notes based on the location in the repo (e.g. notes in `/areas/finance` will automatically get the `#finance` tag)
+- add a new CLI command to support some internal use case or automate import/export
+- extend the VSCode experience to support one's own workflow, e.g. weekly note, templates, extra panels, foam model derived TOC, ... all without having to write/deploy a VSCode extension
+
+## Technical approach
+
+When Foam is loaded it will:
+- check `.foam/plugins` directory.
+	- each directory in there is considered a plugin
+	- the layout of each directory is
+		- `index.js` contains the main info about the plugin, specifically it exports:
```

Comment:
unfortunately this is the case (atm). 
We _could_ depend on typescript and compile them on the fly, but that's a heavy dependency, I didn't think it is worth it at this point (if someone wants to take it on, that'd be great!). 
Not sure if there is another approach that would work to avoid the dependency